android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/TranslationsMiddleware.kt

```diff
@@ -466,6 +471,47 @@ class TranslationsMiddleware(
         )
     }
 
+    /**
+     * Fetches the expected translation model download size assuming the user intends to complete
+     * a translation using detected defaults.
+     *
+     * If detected defaults are available, then this will sets the values on
+     * [SessionState.translationsState] on the [BrowserStore]
+     * via [TranslationsAction.FetchTranslationDownloadSizeAction].
+     *
+     * If no defaults are available, then no action will occur.
+     *
+     * @param context Context to use to dispatch to the store.
+     * @param tabId Tab ID associated with the request.
+     *
+     */
+    private fun requestDefaultModelDownloadSize(
+        context: MiddlewareContext<BrowserState, BrowserAction>,
+        tabId: String,
+    ) {
+        val detectedLanguages = context.store.state.findTab(tabId)
+            ?.translationsState?.translationEngineState?.detectedLanguages
+        val supportedLanguages = context.store.state.translationEngine.supportedLanguages
+        val fromLanguage = detectedLanguages?.documentLangTag?.let { fromTag ->
+            supportedLanguages?.findLanguage(fromTag)
+        }
+        val toLanguage = detectedLanguages?.userPreferredLangTag?.let { toTag ->
+            supportedLanguages?.findLanguage(toTag)
+        }
```

Comment:
I remember in another PR that you have added helper functions that feels similar to what I am seeing here. It is probably not the same as these, but I have a hunch that we might have to do this dance between getting the `detectedLanguages` and `supportedLanguages` in order to get the from/to language in more than one place. Would this be a worthy candidate for a helper function? To be specific, this helper function would get the from/to language from the detected languages.

Reply:
>Would this be a worthy candidate for a helper function? 

Sure, I think so too! Good chance it'll show up again somewhere. Not sure of a good place to globally put it, but I can put a private version of it on this file. One complication is part of the information is on the session store and part is in the browser store, for figuring out something more universal.